### DISCUSSION TITLE
KubernetesExecutor and providing secrets to pods initiated pods

### DISCUSSION THREAD

**Discussion by Miskho** [apache/airflow#23569] [General] [answered]
Hi everyone, 

I am running Airflow using KubernetesExecutor on my Kubernetes cluster. I am using TaskFlow API to define my DAGs like this,
```
executor_config = {
      ...
      k8s.V1Container(
          name="base",
          image="my-dependencies-image"
     ....
}


@dag(
   ...
)
def my_dag():
    @task(executor_config=executor_config)
    def some_task(**kwargs):
    # do some job...
    
dag = my_dag()
```

I am running this on a Kubernetes cluster using official Airflow Helm chart. without some special configurations (I enabled persistent dags and logs and I am using custom image for airflow). I have an environment file with variables which I use to create a secret like this: 
```
kubectl create secret generic my-secret --from-env-file=.env
```

Everything good so far,  the secret and DAGs are both on cluster and running. Now what I would like to accomplish is to isolate this secret from the pods running Airflow (postgresql, scheduler, triggerer, webserver) and the worker-pods which belong to other DAGs, i.e. I don't want to set this in my `values.yaml`:
```
extraEnvFrom: |
 - secretRef:
     name: "my-secret"
```

What I want instead is to be able to somehow "inject" this secret as environment variables only to the pod for this specific DAG.

How can I achieve this?

**Comment by potiuk** [comment, marked as answer]
https://airflow.apache.org/docs/apache-airflow/stable/executor/kubernetes.html#pod-template-file
